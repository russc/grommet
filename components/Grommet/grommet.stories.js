"use strict";

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _grommet = require("grommet");

var _grommetIcons = require("grommet-icons");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return _react.default.createElement(_grommet.Grommet, {
    theme: customTheme
  }, _react.default.createElement(_grommet.Box, {
    pad: "medium"
  }, _react.default.createElement(_grommet.Anchor, {
    icon: _react.default.createElement(_grommetIcons.Add, null),
    label: "Add",
    color: "custom"
  })));
};

(0, _react2.storiesOf)('Grommet', module).add('Theme', function () {
  return _react.default.createElement(Themed, null);
});