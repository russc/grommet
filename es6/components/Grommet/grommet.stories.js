import React from 'react';
import { storiesOf } from '@storybook/react';
import { Grommet, Anchor, Box } from 'grommet';
import { Add } from "grommet-icons/es6/icons/Add";
var customTheme = {
  global: {
    colors: {
      custom: '#cc6633'
    }
  }
};

var Themed = function Themed() {
  return React.createElement(Grommet, {
    theme: customTheme
  }, React.createElement(Box, {
    pad: "medium"
  }, React.createElement(Anchor, {
    icon: React.createElement(Add, null),
    label: "Add",
    color: "custom"
  })));
};

storiesOf('Grommet', module).add('Theme', function () {
  return React.createElement(Themed, null);
});