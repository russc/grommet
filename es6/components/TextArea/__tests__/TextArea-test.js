import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { Grommet } from '../../Grommet';
import { TextArea } from '..';
jest.mock('react-dom');
test('TextArea renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(TextArea, {
    id: "item",
    name: "item"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('TextArea placeholder renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(TextArea, {
    id: "item",
    name: "item",
    placeholder: "placeholder"
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('TextArea plain renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(TextArea, {
    id: "item",
    name: "item",
    plain: true
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('TextArea focusIndicator renders', function () {
  var component = renderer.create(React.createElement(Grommet, null, React.createElement(TextArea, {
    id: "item",
    name: "item",
    focusIndicator: true
  })));
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});