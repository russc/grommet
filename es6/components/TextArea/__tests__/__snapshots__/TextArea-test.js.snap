// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea focusIndicator renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:focus {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: #FD6FFF solid 2px;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
  />
</div>
`;

exports[`TextArea placeholder renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:focus {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: #FD6FFF solid 2px;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    placeholder="placeholder"
  />
</div>
`;

exports[`TextArea plain renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
  border: none;
  width: 100%;
  -webkit-appearance: none;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
  />
</div>
`;

exports[`TextArea renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  border: none;
  -webkit-appearance: none;
  padding: 11px;
  outline: none;
  background: transparent;
  color: inherit;
  font-weight: 600;
  margin: 0;
  border: 1px solid rgba(0,0,0,0.33);
  border-radius: 4px;
  width: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:focus {
  border-color: #FD6FFF;
  box-shadow: 0 0 2px 2px #FD6FFF;
}

.c1:focus > circle,
.c1:focus > ellipse,
.c1:focus > line,
.c1:focus > path,
.c1:focus > polygon,
.c1:focus > polyline,
.c1:focus > rect {
  outline: #FD6FFF solid 2px;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
  />
</div>
`;
